uiowa/uiowa
# Initial work on the Topic Landing Page

As we've gone through our migrations to SiteNow we've been a bit slowed up since we don't have a functional equivalent of a tag-based navigation that we use on many student-facing resource guides on campus. A few currently live examples include:

- https://basicneeds.uiowa.edu/resources/
- https://fsl.uiowa.edu/resources/
- https://offcampus.uiowa.edu/guide/
- https://sds.studentlife.uiowa.edu/campusresources/

This is a template we've been rolling out to our partners' websites and has proved to be fairly popular and we'd like to bring it into the Drupal/SiteNow world in some form.

This is our initial attempt at bringing it into SiteNow. It's currently not using Layout Builder (aside from a few CSS classes) and we'll need some assistance getting it fully brought into the stack.

![topic-landing-preview](https://user-images.githubusercontent.com/472923/130143153-a9e6f7ac-a551-424e-aec6-dccc864aadc4.jpg)

# How to test

You'll need to tag some pages or articles in your local environment with taxonomy terms in order for this template to function properly.

Make sure to run `blt frontend` in order to build the SASS files included in the commit.

Upon installation of the "studentlife_topics" module, a new content type "Topic Landing Page" becomes available. After creating a new "Topic Landing Page" node, you're able to select which tags (or terms) show up in the tag "bubbles" and in the navigation below the search while editing the page.

<img width="916" alt="Screen Shot 2021-08-19 at 3 34 17 PM" src="https://user-images.githubusercontent.com/472923/130141374-0340b458-a272-41a9-86a1-5bbfb4a9141f.png">

A featured image can also be chosen as well, but really, that's it. 



# Stuff we haven't quite figured out yet

- Search is not scoped to the tags selected, it just searches the entire site
- How to fit it into the Layout Builder paradigm, currently this is an "as-is" twig template that can't be pulled apart too much. We'd love to figure out how to generalize it.
- How to make our projects more visible to the team. We're trying, haha.
